subscriber.py

```diff
@@ -0,0 +1,230 @@
+import re
+import boto3
+import os
+import logging
+import cfnresponse
+from typing import Optional
+
+level = os.getenv("log_level", "INFO")
+logging.basicConfig(level=level)
+logger = logging.getLogger()
+logger.setLevel(level)
+
+
+cloudwatch_logs_client = boto3.client("logs")
+lambda_client = boto3.client("lambda")
+
+axiom_cloudwatch_forwarder_lambda_arn = os.getenv(
+    "AXIOM_CLOUDWATCH_FORWARDER_LAMBDA_ARN"
+)
+log_group_names = os.getenv("LOG_GROUP_NAMES", None)
+log_group_prefix = os.getenv("LOG_GROUP_PREFIX", None)
+log_group_pattern = os.getenv("LOG_GROUP_PATTERN", None)
+log_groups_return_limit = 50
+
+
+def build_groups_list(
+    all_groups: list,
+    names: Optional[list] = None,
+    pattern: Optional[str] = None,
+    prefix: Optional[str] = None,
+):
+    # ensure filter params have correct values
+    if not names:
```

Comment:
i don't think you need these if statements, the default args should work fine, right?

Reply:
I think it didn't because the environment variables are empty string not an empty list